### PR TITLE
mcs: add scheduling microservice framework

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #5839.

### What is changed and how does it work?

Add a basic framework for scheduling service.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/mcs/scheduling/server/metrics.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import "github.com/prometheus/client_golang/prometheus"
+
+const (
+	namespace       = "scheduling"
+	serverSubsystem = "server"
+)
+
+var (
+	// Meta & Server info.
+	serverInfo = prometheus.NewGaugeVec(
```
[the remark this answers is not in the document]
Comment (rleungx) [reply pkg/mcs/scheduling/server/metrics.go:26]: They do have

File: pkg/mcs/scheduling/server/config.go [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+	// When a slow store affected more than 30% of total stores, it will trigger evicting.
+	defaultSlowStoreEvictingAffectedStoreRatioThreshold = 0.3
+
+	defaultStoreLimitVersion = "v1"
+	defaultEnableWitness     = false
+	defaultHaltScheduling    = false
+)
+
+// Config is the configuration for the resource manager.
+type Config struct {
```
[the remark this answers is not in the document]
Comment (rleungx) [reply pkg/mcs/scheduling/server/config.go:90]: Yes, this part will be refactored later.

**Comment by rleungx** [comment]
@lhy1024 @JmPotato I just revert the config-related change and do it later. Right now, it's quite simple. I just add a basic framework of microservices. PTAL.

**Review by JmPotato — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**